### ISSUE TITLE
[REQUEST] Add a `disable` argument for progress bar

### ISSUE THREAD

**Issue by Mr-Milk** [willmcgugan/rich#449]
**How would you improve Rich?**

In tqdm there is a `disable` argument to disable the progress bar.

```
disable: bool, optional
Whether to disable the entire progress bar wrapper [default: False]. If set to None, disable on non-TTY.
```

This is very useful for some applications to control output verbosity. Otherwise, an if statement is needed which makes the code dirty.

Before using `disable`

```python
from rich.progress import track

if verbosity:
    for i in range(100):
        print(i)
else:
    for i in track(range(100)):
        print(i)
```

If we have `disable`

```python
from rich.progress import track

for i in track(range(100), disable=verbosity):
    print(i)
```

**Comment by willmcgugan** [comment]
Seems reasonable. I think I can add that to the next release.

**Comment by willmcgugan** [comment]
I've added a disable parameter to Progress in 9.3.0, on PyPi now.

**Comment by Mr-Milk** [comment]
Thanks for developing such a great package. 

I tried it but doesn't seem to work, am I making some mistakes?

Also, do you have a future plan to add the `disable` to the `track` function? I think `track` is used more than a `Progress` context.

<img width="500" alt="example" src="https://user-images.githubusercontent.com/23433306/100826026-6937c980-3494-11eb-8e46-b9f3098c5250.png">

**Comment by willmcgugan** [comment]
I forgot to implement it in Jupyter. I'll add that to the next release.

**Comment by kdcloverkid** [comment]
> **How would you improve Rich?**
> 
> In tqdm there is a `disable` argument to disable the progress bar.
> 
> ```
> disable: bool, optional
> Whether to disable the entire progress bar wrapper [default: False]. If set to None, disable on non-TTY.
> ```
> 
> This is very useful for some applications to control output verbosity. Otherwise, an if statement is needed which makes the code dirty.
> 
> Before using `disable`
> 
> ```python
> from rich.progress import track
> 
> if verbosity:
>     for i in range(100):
>         print(i)
> else:
>     for i in track(range(100)):
>         print(i)
> ```
> 
> If we have `disable`
> 
> ```python
> from rich.progress import track
> 
> for i in track(range(100), disable=verbosity):
>     print(i)
> ```https://github.com/willmcgugan/rich/issues/449#issue-752570112

**Comment by kdcloverkid** [comment]
Hmm

**Comment by willmcgugan** [comment]
Fixed in 9.4.0
